### PR TITLE
Adding in functionality to download from a private or public mirror

### PR DESCRIPTION
**Pull request by davidjoyce-okta**
- Adding in functionality to download from a private or public mirror directly if version not available from Oracle.
- General clean-up, restructuring and attempting to ensure consistent style.
- Bug fixes and replacing deprecated ansible commands.

Please review: @srsp

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,11 +7,11 @@ srsp.oracle-java for Ansible Galaxy
 
 Role name in Ansible Galaxy: **[srsp.oracle-java](https://galaxy.ansible.com/srsp/oracle-java/)**
 
-This Ansible role has the following features for Oracle JDK:
+This Ansible role has the following features related to the Oracle JDK:
 
- - Install JDK 8 or 10 in the latest version.
- - Install optional Java Cryptography Extensions (JCE). Only for JDK 8, because it is no longer needed for JDKs > 8.
- - Install for CentOS, Debian/Ubuntu, SUSE, and Mac OS X families.
+ - Install the latest version of Oracle JDK 8 or 10.
```
Comment (albertli-okta) [inline README.md:12]: For us, maybe we want 7 or 8?

Comment (albertli-okta) [inline README.md:12]: For us, maybe we want at least 7, 8 and 10?

Comment (albertli-okta) [inline README.md:12]: For us, maybe we want at least 7, 8 and 10?

Comment (albertli-okta) [inline README.md:12]: For us, maybe we want at least 7, 8 and 10?

Comment (albertli-okta) [inline README.md:12]: For us, maybe we want at least 7, 8 and 10?

Comment (albertli-okta) [inline README.md:12]: we need add least 7,8 and 10?

Comment (srsp) [reply README.md:12]: I decided to drop 7 a long time ago, because it eol'ed a long time ago.

**Comment by srsp** [comment]
@davidjoyce-okta Thank you for this huge PR! I took a first look and it seems all nice and fine. You were very thorough. I'll take a closer look today and give you feedback.

File: README.md (hunk 1 above)
Comment (srsp) [reply README.md]: I decided to drop 7 a long time ago, because it eol'ed a long time ago.

**Comment by srsp** [comment]
@davidjoyce-okta Thank you for this huge PR! I took a first look and it seems all nice and fine. You were very thorough. I'll take a closer look today and give you feedback.

**Comment by srsp** [comment]
@davidjoyce-okta Thank you for this huge PR! I took a first look and it seems all nice and fine. You were very thorough. I'll take a closer look today and give you feedback.

File: README.md (hunk 1 above)
Comment (srsp) [reply README.md]: I decided to drop 7 a long time ago, because it eol'ed a long time ago.

Comment (srsp) [reply README.md]: I decided to drop 7, because it eol'ed a long time ago.

Comment (srsp) [reply README.md]: I decided to drop 7, because it eol'ed a long time ago.

Comment (srsp) [reply README.md:12]: 10 is probably not downloadable anymore without Oracle account. 11 should be supported, but I did not yet find the time to support it.

**Pull request merged**